File: python/minimum_number_of_steps_to_make_two_strings_anagram.py
# no.1347
class Solution:
    def minSteps(self, s: str, t: str) -> int:
        move_map = {}
        length = len(s)
        for i in range(length):
            move_map[s[i]] = move_map.get(s[i], 0) + 1
            move_map[t[i]] = move_map.get(t[i], 0) - 1
        ans = 0
        for c in move_map:
            if move_map[c] > 0:
                ans += move_map[c]
        return ans

File: C++/minimum_number_of_steps_to_make_two_strings_anagram.cpp
// no. 1347
#include <map>
#include <string>
using namespace std;

class Solution {
public:
  int minSteps(string s, string t) {
    int length = s.length();
    map<char, int> move_map;
    for (int i = 0; i < length; i++) {
      move_map[s[i]]++;
      move_map[t[i]]--;
    }
    int ans = 0;
    for (map<char, int>::iterator it = move_map.begin(); it != move_map.end();
         it++) {
      if (it->second > 0) {
        ans += it->second;
      }
    }
    return ans;
  }
};
